pkg/auth/authz_handler.go

```diff
@@ -0,0 +1,139 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/pkg/ocm"
+	"github.com/openshift/assisted-service/restapi"
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	bareMetalClusterResource string = "BareMetalCluster"
+	amsActionCreate          string = "create"
+	clustersPathRegex        string = "clusters/.+"
+	capabilityName           string = "bare_metal_installer_admin"
+	capabilityType           string = "Account"
+)
+
+// CreateAuthorizer returns Authorizer if auth is enabled
+func (a *AuthHandler) CreateAuthorizer() func(*http.Request) error {
+	if !a.EnableAuth {
+		return func(*http.Request) error {
+			return nil
```

Comment:
If not enabled, return a system admin user so the rest of the code can work correctly